CorpusCallosum/shape/cc_thickness.py

```diff
@@ -0,0 +1,543 @@
+
+    Parameters
+    ----------
+    contour_with_thickness : list[np.ndarray]
+        List containing [contour_points, thickness_values].
+    point : np.ndarray
+        2D point to insert, shape (2,).
+    thickness_value : float
+        Thickness value corresponding to the point.
+    get_index : bool, optional
+        If True, return the index where point was inserted, by default False.
+
+    Returns
+    -------
+    tuple[list[np.ndarray], int] or list[np.ndarray]
+        If get_index is True:
+            - Updated contour_with_thickness.
+            - Index where point was inserted.
+        If get_index is False:
+            - Updated contour_with_thickness.
+    """
+    # Find closest edge for the point
+    edge_idx = find_closest_edge(point, contour_with_thickness[0])
+
+    # Insert point between edge endpoints
+    contour_with_thickness[0] = np.insert(
+        contour_with_thickness[0], edge_idx + 1, point, axis=0
+    )
+    contour_with_thickness[1] = np.insert(
+        contour_with_thickness[1], edge_idx + 1, thickness_value
+    )
+
+    if get_index:
+        return contour_with_thickness, edge_idx + 1
+    else:
+        return contour_with_thickness
+
+
+def make_mesh_from_contour(
+    contour_2d: np.ndarray,
+    max_volume: float = 0.5,
+    min_angle: float = 25,
+    verbose: bool = False
+) -> tuple[np.ndarray, np.ndarray]:
+    """Create a triangular mesh from a 2D contour.
+
+    Parameters
+    ----------
+    contour_2d : np.ndarray
+        Array of shape (N, 2) containing contour points.
+    max_volume : float, optional
+        Maximum triangle area, by default 0.5.
+    min_angle : float, optional
+        Minimum angle in triangles (degrees), by default 25.
+    verbose : bool, optional
+        Whether to print mesh generation info, by default False.
+
+    Returns
+    -------
+    tuple[np.ndarray, np.ndarray]
+        - mesh_points : Array of shape (M, 2) containing mesh vertices.
+        - mesh_trias : Array of shape (K, 3) containing triangle indices.
+
+    Notes
+    -----
+    Uses meshpy.triangle to create a constrained Delaunay triangulation
+    of the contour. The contour must not have duplicate points.
+    """
+
+    facets = np.vstack(
+        (
+            np.arange(len(contour_2d)),
+            ((np.arange(len(contour_2d)) + 1) % len(contour_2d)),
+        )
+    ).T
+
+    # plot vertices and facets
+    # import matplotlib.pyplot as plt
+    # fig, ax = plt.subplots(figsize=(10, 8))
+    # ax.scatter(contour_2d[:,0], contour_2d[:,1], label='Contour')
+    # ax.plot(contour_2d[:,0], contour_2d[:,1], 'k-', label='Contour')
+    # ax.plot(contour_2d[facets[:,0],0], contour_2d[facets[:,0],1], 'r-', label='Facets')
+    # plt.show()
+
+    # use meshpy to create mesh
+    info = triangle.MeshInfo()
+    info.set_points(contour_2d)
+    info.set_facets(facets)
+    # NOTE: crashes if contour has duplicate points !!
+    mesh = triangle.build(
```

Comment:
It seems to me, this is currently the only dependency for meshpy of the FastSurfer project. For what do we need this here and does lapy not support that operation?

It seems to be some meshing operation?

Reply:
It creates a planar triangle mesh from a closed polygon contour. Lapy does not create meshes, it operates on meshes.